how-to/integrate-with-ms365-microflow-basic/client/src/provider.ts

```diff
@@ -0,0 +1,57 @@
+import type OpenFin from "@openfin/core";
+import { Home, Integrations } from "@openfin/workspace";
+import { init } from "@openfin/workspace-platform";
+import { type WorkflowIntegration } from "@openfin/workspace-platform/client-api/src/shapes/integrations";
+import { type CustomSettings } from "./shapes";
+
+const PLATFORM_ICON = "http://localhost:8080/favicon.ico";
+
+window.addEventListener("DOMContentLoaded", async () => {
+	// The DOM is ready so initialize the platform
+	await initializeWorkspacePlatform();
+});
+
+/**
+ * Initialize the workspace platform.
+ */
+async function initializeWorkspacePlatform(): Promise<void> {
+	console.log("Initializing workspace platform");
+	const customSettings = await getManifestCustomSettings();
+	const integrations: WorkflowIntegration[] = [];
+	if (customSettings.ms365) {
+		const msft365WorkflowIntegration = new Integrations.Microsoft365WorkflowIntegration({
+			connect: {
+				clientId: customSettings.ms365.clientId,
+				redirectUri: customSettings.ms365.redirectUri,
+				tenantId: customSettings.ms365.tenantId
+			}
+		});
+		integrations.push(msft365WorkflowIntegration as unknown as WorkflowIntegration);
```

Comment:
Why is this cast needed?

Reply:
It is needed for now until the type export is updated